libflux/flux-core/src/semantic/bootstrap.rs

```diff
@@ -35,32 +36,89 @@ pub struct Error {
+    Builtin(Box<ValueDoc>),
+    /// Option represents documentation for a option value exposed from a package.
+    Option(Box<ValueDoc>),
+    /// Function represents documentation for a function value exposed from a package.
+    Function(Box<FunctionDoc>),
+}
+
+/// PackageDoc represents the documentation for a package and its sub packages
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PackageDoc {
+    /// the name of the comments package
     pub name: String,
-    /// the doc
-    pub doc: String,
-    /// the values of the comments
-    pub values: Vec<DocValue>,
+    /// the headline of the package
+    pub headline: String,
+    /// the description of the package
+    pub description: Option<String>,
+    /// the members are the values and funcitons of a package
+    pub members: HashMap<String, Doc>,
 }
 
-/// DocValue represents the documentation for a single value within a package.
+/// ValueDoc represents the documentation for a single value within a package.
 /// Values include options, builtins or any variable assignment within the top level scope of a
 /// package.
 #[derive(Debug, Serialize, Deserialize)]
-pub struct DocValue {
-    /// the path to the package
-    pub pkgpath: String,
-    /// the name of the package
+pub struct ValueDoc {
+    /// the name of the value
+    pub name: String,
+    /// the headline of the value
+    pub headline: String,
+    /// the description of the value
+    pub description: Option<String>,
+    /// the type of the value
+    pub flux_type: String,
+}
+
+/// FunctionDoc represents the documentation for a single Function within a package.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct FunctionDoc {
+    /// the name of the function
     pub name: String,
-    /// the doc
-    pub doc: String,
-    /// the type of the comments
-    pub typ: String,
+    /// the headline of the function
+    headline: String,
+    /// the description of the function
+    description: String,
+    /// the parameters of the function
+    parameters: Vec<ParameterDoc>,
+    /// the type of the function
+    flux_type: String,
+}
+
+/// Implementation block for FunctionDoc to house future "FunctionDod" methods
+impl FunctionDoc {
+    /// Constructs a new FunctionDoc with the given name, headline, and flux type
+    fn new_with_args(name: String, doc: String, typ: String) -> FunctionDoc {
+        FunctionDoc {
+            name,
+            headline: doc,
+            description: "".to_string(),
+            parameters: vec![],
+            flux_type: typ,
+        }
+    }
+}
+
+/// ParameterDoc represents the documentation for a single parameter within a function.
+#[derive(Debug, Serialize, Deserialize)]
+struct ParameterDoc {
+    /// the name of the parameter
+    name: String,
+    /// the headline of the parameter
+    headline: String,
+    /// the description of the parameter
+    description: Option<String>,
+    /// a boolean indicating if the parameter is required
+    required: bool,
```

Comment:
Do we also need a boolean that says whether the parameter is the stream argument?

Reply:
That might be nice to add in the future, but for now I don't think it's necessary.